assets/src/edit-story/app/user/userProvider.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useCallback, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../api';
+import Context from './context';
+
+function UserProvider({ children }) {
+  const [currentUser, setCurrentUser] = useState({});
+  const {
+    actions: { getCurrentUser: _getCurrentUser },
+  } = useAPI();
+
+  const getCurrentUser = useCallback(async () => {
+    if (Object.entries(currentUser).length) {
+      return currentUser;
+    }
+    const _currentUser = await _getCurrentUser();
+    setCurrentUser(_currentUser);
+
+    return _currentUser;
+  }, [_getCurrentUser, currentUser]);
+
+  const state = {
+    actions: {
+      getCurrentUser,
+    },
+  };
```

Comment:
This seems very unintuitive.

I'd rather use `useEffect` here instead of a function that may or may not issue an API request.

Something like that:

```js
const [currentUser, setCurrentUser] = useState({});
const { actions: { getCurrentUser } } = useAPI();
useEffect(() => {
  if (!Object.keys(currentUser).length) {
    setCurrentUser(await getCurrentUser());
  }
}, [currentUser, getCurrentUser]);

const state = {
  state: {
    currentUser,
  }
  actions: {
    getCurrentUser,
  },
};
```

Bonus: `canTranscodeFile` doesn't need to be async.

Reply:
Yeah, I did that originally, but I was worried about users dragging a file in editor, the second it loads and this can code doesn't have time to take effect.

Reply:
Sounds like an unlikely scenario, especially with the REST API preloading. This code will run super early.

If we're really concerned about this after testing we could just disable uploading until the user data is loaded.

Reply:
Hm, I don't mind it fetching conditionally - it returns a promise that'll resolve whenever the current user is known, which might be instant. Works for me, actually :)

Reply:
But there is a need to update the current user. So this won't work because it's only fetched once and once only.

Reply:
Changed as requested.